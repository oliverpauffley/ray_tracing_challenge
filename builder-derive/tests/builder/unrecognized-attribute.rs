use builder_derive::Builder;

#[derive(Builder, Debug, PartialEq)]
struct Command {
    name: String,
    shapes: Vec<String>,
    properties: Option<i64>,
}

#[derive(Builder, Debug, PartialEq)]
struct File {
    name: String,
    #[builder(eac = "location")]
    locations: Vec<String>,
}

fn main() {
    let builder = Command::builder();
    assert!(builder.name.is_none());
    assert!(builder.shapes.is_none());
    assert!(builder.properties.is_none());
}
