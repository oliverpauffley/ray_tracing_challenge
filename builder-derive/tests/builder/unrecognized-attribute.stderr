error: expected `builder(each = "...")`
  --> tests/builder/unrecognized-attribute.rs:13:7
   |
13 |     #[builder(eac = "location")]
   |       ^^^^^^^^^^^^^^^^^^^^^^^^^
